src/nodedef.cpp

```diff
@@ -809,14 +810,22 @@ void CNodeDefManager::updateTextures(IGameDef *gamedef,
 				f->drawtype = NDT_ALLFACES;
 				f->solidness = 0;
 				f->visual_solidness = 1;
+				if (f->waving == 1)
+					material_type = TILE_MATERIAL_WAVING_LEAVES;
 			} else {
-				f->drawtype = NDT_NORMAL;
-				f->solidness = 2;
-				for (u32 i = 0; i < 6; i++)
-					tiledef[i].name += std::string("^[noalpha");
+				if (plantlike_leaves) {
+					f->visual_scale = 1.1892;
```

Comment:
1 means that the height and the width of the polygon is the node height (1m).
lf the polygon's width is 1, there would be visual spaces between the nodes because the polygon is diagonal seen from above, and so the the visual space size would be 1-sqrt(1*1/2)